flowey/flowey_lib_hvlite/src/_jobs/check_openvmm_hcl_size.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Compares the size of the OpenHCL binary in the current PR with the size of the binary from the last successful merge to main.
+
+use crate::artifact_openhcl_igvm_from_recipe_extras;
+use crate::build_openhcl_igvm_from_recipe;
+use crate::build_openhcl_igvm_from_recipe::OpenhclIgvmRecipe;
+use crate::build_openvmm_hcl;
+use crate::build_openvmm_hcl::OpenvmmHclBuildParams;
+use crate::build_openvmm_hcl::OpenvmmHclBuildProfile::OpenvmmHclShip;
+use crate::run_cargo_build::common::CommonArch;
+use crate::run_cargo_build::common::CommonTriple;
+use flowey::node::prelude::*;
+use flowey_lib_common::download_gh_artifact;
+use flowey_lib_common::gh_workflow_id;
+use flowey_lib_common::git_merge_commit;
+
+flowey_request! {
+    pub struct Request {
+        pub target: CommonTriple,
+        pub done: WriteVar<SideEffect>,
+        pub pipeline_name: String,
+    }
+}
+
+new_simple_flow_node!(struct Node);
+
+impl SimpleFlowNode for Node {
+    type Request = Request;
+
+    fn imports(ctx: &mut ImportCtx<'_>) {
+        ctx.import::<crate::build_xtask::Node>();
+        ctx.import::<crate::git_checkout_openvmm_repo::Node>();
+        ctx.import::<download_gh_artifact::Node>();
+        ctx.import::<git_merge_commit::Node>();
+        ctx.import::<gh_workflow_id::Node>();
+        ctx.import::<build_openhcl_igvm_from_recipe::Node>();
+        ctx.import::<build_openvmm_hcl::Node>();
+        ctx.import::<artifact_openhcl_igvm_from_recipe_extras::publish::Node>();
+    }
+
+    fn process_request(request: Self::Request, ctx: &mut NodeCtx<'_>) -> anyhow::Result<()> {
+        let Request {
+            target,
+            done,
+            pipeline_name,
+        } = request;
+
+        let xtask = ctx.reqv(|v| crate::build_xtask::Request {
+            target: target.clone(),
+            xtask: v,
+        });
+        let openvmm_repo_path = ctx.reqv(crate::git_checkout_openvmm_repo::req::GetRepoDir);
+
+        let gh_token = ctx.get_gh_context_var().global().token();
+
+        let built_openvmm_hcl = ctx.reqv(|v| build_openvmm_hcl::Request {
+            build_params: OpenvmmHclBuildParams {
+                target: target.clone(),
+                profile: OpenvmmHclShip,
+                features: (OpenhclIgvmRecipe::X64)
+                    .recipe_details(OpenvmmHclShip)
+                    .openvmm_hcl_features,
+                no_split_dbg_info: false,
+            },
+            openvmm_hcl_output: v,
+        });
+
+        let file_name = match target.common_arch().unwrap() {
```

Comment:
this is the kind of logic you really want to be in the `artifact_openvmm_hcl_sizecheck::resolve` node. i.e: you have tight locality between the two bits of code that determine the "shape" of artifact. in addition, having the resolve node gives you a type-safe handle to the artifact content when you download it, vs. needing to do this sort of ad-hoc "unpacking" inline.

if you don't do that, then when the time comes to add more stuff into that artifact, you'll have zero compiler assistance in tracking down what code needs to be updated to support the new artifact.